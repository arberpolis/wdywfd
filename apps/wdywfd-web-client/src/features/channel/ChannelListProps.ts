import { PropsWithChildren, } from "react";
import { ChannelType } from "./ChannelType";

export interface ChannelListProps extends PropsWithChildren<{}> {
    readonly error?: boolean
    readonly loading: boolean
    readonly type: ChannelType
}   // ChannelProps

