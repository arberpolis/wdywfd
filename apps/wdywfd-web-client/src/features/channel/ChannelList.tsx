import React from 'react'
import { ChannelListProps } from './ChannelListProps'

const ChannelList = ({ children, error, loading, type }: ChannelListProps) => {
  return (
    <div className="channel-list">
      <div className="channel-list-header">
        <div className="channel-list-header-title">
          <h3>{type}</h3>
        </div>
      </div>
      <div className="channel-list-body">
        {loading && <div className="channel-list-loading">Loading...</div>}
        {error && <div className="channel-list-error">Error!</div>}
        {children}
      </div>
    </div>
  )
}

export default ChannelList
