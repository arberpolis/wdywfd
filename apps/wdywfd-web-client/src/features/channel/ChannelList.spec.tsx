import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import ChannelList from './ChannelList';
import { ChannelListProps } from './ChannelListProps';

describe('ChannelList', () => {
    let sut: ReactTestRenderer;

    let props: ChannelListProps = {
        children: [(
            <div className="channel" key="channel-1">Channel 1</div>
        ), (
            <div className="channel" key="channel-2">Channel 2</div>
        )],
        loading: false,
        type: 'team',
    }

    beforeEach(() => {
        sut = renderer.create(<ChannelList { ...props } />);
    });

    it('is defined', () => {
        expect(sut).toBeDefined();
    });
    it('renders correctly', () => {
        expect(sut.toJSON()).toMatchSnapshot();
    });
});
