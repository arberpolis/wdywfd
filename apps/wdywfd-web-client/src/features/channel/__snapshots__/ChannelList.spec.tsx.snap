// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList should render correctly 1`] = `
<div
  className="channel-list"
>
  <div
    className="channel-list-header"
  >
    <div
      className="channel-list-header-title"
    >
      <h3>
        team
      </h3>
    </div>
  </div>
  <div
    className="channel-list-body"
  >
    <div
      className="channel"
    >
      Channel 1
    </div>
    <div
      className="channel"
    >
      Channel 2
    </div>
  </div>
</div>
`;
